src/odemis/acq/test/fastem_test.py

```diff
@@ -189,23 +236,23 @@ def test_get_poly_field_indices(self):
         roc_2 = fastem.FastEMROC("roc_2", coordinates)
         roc_3 = fastem.FastEMROC("roc_3", coordinates)
         roa_name = "test_megafield_id"
-        roa = fastem.FastEMROA(roa_name,
-                               roc_2,
-                               roc_3,
-                               self.asm,
-                               self.multibeam,
-                               self.descanner,
-                               self.mppc,
-                               overlap=0.0)
-
-        # Index (1,0) is ignored because of the step over effect the algorithm can have with very small corners.
+        roa = fastem.FastEMROA(shape=MockEditableShape(),
+                        main_data=self.main_data,
+                        overlap=0.0,
+                        name=roa_name,
+                        slice_index=0)
+        roa.roc_2.value = roc_2
+        roa.roc_3.value = roc_3
+        roa.polygon_shape = Polygon(polygon)
+
         expected_indices = [(0, 0),
-                            (0, 1), (1, 1), (2, 1),
-                            (0, 2), (1, 2), (2, 2), (3, 2),
-                            (0, 3), (1, 3), (2, 3), (3, 3), (4, 3)]  # (col, row)
+                            (0, 1), (1, 1),
+                            (0, 2), (1, 2), (2, 2),
+                            (0, 3), (1, 3), (2, 3), (3, 3),
+                            (0, 4), (1, 4), (2, 4), (3, 4)]
```

Comment:
Are you sure this is correct? Because on line 227 it is defined as x_field = 5 and y_fields = 4, so I would expect 5 columns and 4 rows. Now you have 4 columns and 5 rows.

Reply:
Good catch, the culprit is the `polygon` points which are (y, x) and hence the flip

Reply:
Then I would suggest to correct it on line 236